### DISCUSSION TITLE
`dist` folder and pre-built, minified `.js` bundle missing?

### DISCUSSION THREAD

**Discussion by drinkspiller** [fabricjs/fabric.js#10160] [General]
Just a heads up — the `/dist` folder and a pre-built, minified .js bundle within it appears to be missing in the repo root. 

In addition to it being convenient to have easy access to a pre-build bundle, a few of the codepens referenced in the documentation are unable to run because they expect a built bundle to be in `/dist`:

![image](https://github.com/user-attachments/assets/fd83d01e-f9c3-4d59-8db7-ca3c67662036)

![image](https://github.com/user-attachments/assets/6165e007-c4e0-4c52-920e-cd737986d6d9)

**Comment by drinkspiller** [comment]
The ideal would be to have the dist folder contain two pre-built version:
1. a "raw" version (pre-built, bundled into a single file, but not minified)
2. a "min" version (pre-built, bundled into a single file, and minified)


It looks like these do get built as part of the [NPM package publish](https://github.com/fabricjs/fabric.js/blob/master/publish-next.js#L9):
![image](https://github.com/user-attachments/assets/77b2db27-f854-464d-bc6c-775f30b081fc)

**Comment by asturur** [comment]
So the issue here was just not creating useless diff on github.
That is why it was removed.
Re-enabling it would mean remove a line in .gitignore and run npm run build every time i do a release.

that is all we are talking about.

I still ask myself if it make sense.
I understand you have constrain at work  and why you need this, but is this the right use of the tool for github?

(please consider that fabricjs.com is outdated and is being replaced by fabricjs.github.io that as some point will be copied over fabricjs.com)

**Comment by drinkspiller** [comment]
> is this the right use of the tool for github

It certainly is not uncommon for some of the largest, most depped/downloaded projects on Github to include a `/dist` folder which prebuilt binaries in the repo:

- https://github.com/moment/moment
- https://github.com/caolan/async
- https://github.com/axios/axios
- https://github.com/jquery/jquery

**Comment by asturur** [comment]
I don't have an issue re-enabling those on release.
My personal opinion is that there isn't a best practice or nor the 4 pulled libraries should be taken in account as notable examples.
So we can just do what is comfortable for developers, adding the dist files on the repo won't hurt anyone apart github servers, so i m fine with it.

**Comment by asturur** [comment]
Do you need types to be there?
or maps?

**Comment by drinkspiller** [comment]
Yes, types would be great, thank you. Maps nice but not required. Cheers!
